flink-table-store-common/src/main/java/org/apache/flink/table/store/file/predicate/AlwaysFalse.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.file.predicate;
+
+import org.apache.flink.table.store.format.FieldStats;
+import org.apache.flink.table.types.logical.LogicalType;
+
+import java.util.Optional;
+
+/** Return false for all values. */
+public class AlwaysFalse extends LeafUnaryFunction {
+    public static final AlwaysFalse INSTANCE = new AlwaysFalse();
+
+    private AlwaysFalse() {}
+
+    @Override
+    public Optional<LeafFunction> negate() {
+        return Optional.of(AlwaysTrue.INSTANCE);
+    }
+
+    @Override
+    public boolean test(LogicalType type, Object value) {
```

Comment:
We can have better way to `AlwaysFalse` and `AlwaysTrue`, because they don't need to have field reference. But this requires more refactor.

Can you add TODO in the code and create a JIRA for this?

Reply:
Added https://issues.apache.org/jira/browse/FLINK-30227